contrib/golang/router/cluster_specifier/source/cgo.cc

```diff
@@ -0,0 +1,55 @@
+#include "contrib/golang/router/cluster_specifier/source/golang_cluster_specifier.h"
+
+namespace Envoy {
+namespace Router {
+namespace Golang {
+
+//
+// These functions should only be invoked in the current Envoy worker thread.
+//
+
+enum GetHeaderResult {
+  Mising = 0,
+  Found = 1,
+};
+
+absl::string_view referGoString(void* str) {
+  if (str == nullptr) {
+    return "";
+  }
+  auto goStr = reinterpret_cast<GoString*>(str);
+  return absl::string_view(goStr->p, goStr->n); // NOLINT(modernize-return-braced-init-list)
+}
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+int envoyGoClusterSpecifierGetHeader(unsigned long long header_ptr, void* key, void* value) {
```

Comment:
nit: it would be better if there are some more detailed comments to tell the reader the meaning of this func and parameters.